test/e2e/specs/onboarding/signup__with-theme-premium.ts

```diff
@@ -0,0 +1,207 @@
+	const planName = 'Premium';
+	let themeSlug: string | null = null;
+
+	const testUser = DataHelper.getNewTestUser( {
+		usernamePrefix: 'ftmepremium',
+	} );
+
+	let page: Page;
+	let newUserDetails: NewUserResponse;
+	let newSiteDetails: NewSiteResponse;
+
+	beforeAll( async () => {
+		page = await browser.newPage();
+	} );
+
+	describe( 'Signup', function () {
+		let cartCheckoutPage: CartCheckoutPage;
+		let signupPickPlanPage: SignupPickPlanPage;
+
+		beforeAll( async function () {
+			await BrowserManager.setStoreCookie( page, { currency: 'USD' } );
+		} );
+
+		it( 'Navigate to Theme Showcase', async function () {
+			const themesPage = new ThemesPage( page );
+			await themesPage.visitShowcase();
+		} );
+
+		it( 'Selects a Premium theme', async function () {
+			await page
+				.locator( 'div.theme-card:has(div.premium-badge):not(div.is-marketplace)' )
+				.first()
+				.click();
+		} );
+
+		it( 'Navigate to Signup page', async function () {
+			const themeDetailsPage = new ThemesDetailPage( page );
+
+			const pageMatch = new URL( page.url() ).pathname.match( 'theme/(.*)/?' );
+
+			themeSlug = Array.isArray( pageMatch ) ? pageMatch[ 1 ] : null;
+
+			await themeDetailsPage.pickThisDesign();
+		} );
+
+		it( 'Sign up as new user', async function () {
+			const userSignupPage = new UserSignupPage( page );
+			newUserDetails = await userSignupPage.signupSocialFirstWithEmail( testUser.email );
+		} );
+
+		it( 'Skip domain selection', async function () {
+			const domainSearch = new DomainSearchComponent( page );
+
+			await domainSearch.search( testUser.siteName );
+			await domainSearch.selectDomain( `${ testUser.siteName }.wordpress.com` );
+		} );
+
+		it( `Select WordPress.com ${ planName } plan`, async function () {
+			signupPickPlanPage = new SignupPickPlanPage( page );
+			newSiteDetails = await signupPickPlanPage.selectPlan( planName );
+		} );
+
+		it( 'See secure payment', async function () {
+			cartCheckoutPage = new CartCheckoutPage( page );
+			await cartCheckoutPage.validateCartItem( `WordPress.com ${ planName }` );
+		} );
+
+		it( 'Apply coupon', async function () {
+			await cartCheckoutPage.enterCouponCode( SecretsManager.secrets.testCouponCode );
+		} );
+
+		it( 'Enter billing and payment details', async function () {
+			const paymentDetails = DataHelper.getTestPaymentDetails();
+			await cartCheckoutPage.enterBillingDetails( paymentDetails );
+			await cartCheckoutPage.enterPaymentDetails( paymentDetails );
+		} );
+
+		it( 'Make purchase', async function () {
+			await cartCheckoutPage.purchase( { timeout: 90 * 1000 } );
+		} );
+
+		it( 'Skip business plan upsell', async function () {
+			const selector = 'button[data-e2e-button="decline"]';
+			await page.waitForSelector( selector );
+
+			const locator = page.locator( selector );
+
+			try {
+				await locator.click( { timeout: 2 * 1000 } );
+			} catch {
+				// noop
+			}
```

Comment:
I'm not really sure what is happening here. Why are we trying to click a button that we've already waited for? Is this button possibly disabled or not rendered at all?

Reply:
Good point. This test was built in top of another test. [There](https://github.com/Automattic/wp-calypso/blob/trunk/test/e2e/specs/onboarding/lifecycle__signup-onboarding-launch-cancel.ts#L128), we have this condition because we display the upsell for Business on Premium purchases. We probably catch it because we might cover non-premium purchases where the upsell is not present.

However, in our case, we always test a Premium subscription, so I guess we can remove the try-catch.

Reply:
Nice! Let's also remove the `waitForSelecto` line above - the `click()` waits until the element is clickable.